api_tests/e2e/rfc003/eth-erc20_btc/refund.ts

```diff
@@ -0,0 +1,210 @@
+            actor: bob,
+            action: ActionKind.Accept,
+            requestBody: {
+                beta_ledger_refund_identity: bob.wallet.eth().address(),
+                alpha_ledger_redeem_identity: bobFinalAddress,
+            },
+            state: (state: any) => state.communication.status === "ACCEPTED",
+        },
+        {
+            actor: alice,
+            action: ActionKind.Deploy,
+            state: (state: any) => state.alpha_ledger.status === "Deployed",
+        },
+        {
+            actor: alice,
+            action: ActionKind.Fund,
+            state: (state: any) => state.alpha_ledger.status === "Funded",
+            test: {
+                description:
+                    "[alice] Should have less alpha asset after the funding",
+                callback: async () => {
+                    let erc20BalanceAfter = await ethereum.erc20Balance(
+                        alice.wallet.eth().address(),
+                        tokenContractAddress
+                    );
+                    erc20BalanceAfter
+                        .lt(aliceInitialErc20)
+                        .should.be.equal(true);
+                },
+            },
+        },
+        {
+            actor: bob,
+            action: ActionKind.Fund,
+            state: (state: any) =>
+                state.alpha_ledger.status === "Funded" &&
+                state.beta_ledger.status === "Funded",
+        },
+        {
+            actor: bob,
+            test: {
+                description: "[bob] Is waiting for beta htlc to expire",
+                callback: async () => {
+                    while (Date.now() / 1000 < betaExpiry) {
+                        await sleep(200);
+                    }
+                },
+            },
+        },
+        {
+            actor: bob,
+            action: ActionKind.Refund,
+            uriQuery: { address: bobRefundAddress, fee_per_byte: 20 },
+            state: (state: any) => state.beta_ledger.status === "Refunded",
+            test: {
+                description:
+                    "[bob] Should have received the beta asset after the refund",
+                callback: async (body: any) => {
+                    let refundTxId = body.state.beta_ledger.refund_tx;
+
+                    let satoshiReceived = await bitcoin.getFirstUtxoValueTransferredTo(
+                        refundTxId,
+                        bobRefundAddress
+                    );
+                    const satoshiExpected = betaAssetQuantity - betaMaxFee;
+
+                    satoshiReceived.should.be.at.least(satoshiExpected);
+                },
+            },
+        },
+        {
+            actor: alice,
+            test: {
+                description: "[alice] Is waiting for alpha htlc to expire",
+                callback: async () => {
+                    while (Date.now() / 1000 < alphaExpiry) {
+                        await sleep(200);
+                    }
+                },
```

Comment:
❌ This thing should be unnecessary. The action should be self descriptive so that when you say you want to do `ActionKind.Refund`, the framework should wait until the time specified in the action and then execute it.